### PR TITLE
SAK-40339 Added associated data to assignment site import

### PR DESCRIPTION
**Pull request by adrianfish**

### CODE REVIEW

**Comment by adrianfish** [comment]
Screen Shot 2019-05-16 at 11 10 57

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 1]
```diff
@@ -4053,4 +4141,57 @@ private String getReviewError(Long status){
...
+                toCategoryDefinitions = new ArrayList<>();
+            }
+
+            if (!toCategoryDefinitions.stream().anyMatch(cd -> cd.getName().equals(categoryName))) {
+                // The category doesn't exist yet
+                CategoryDefinition fromCategoryDefinition
+                    = gradebookService.getCategoryDefinitions(fromGradebookId)
+                        .stream()
+                        .filter(cd -> cd.getName().equals(categoryName))
+                            .findFirst().get();
```
Comment (ern) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: if order is not an issue try findAny()

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 2]
```diff
@@ -4053,4 +4141,57 @@ private String getReviewError(Long status){
...
+                toGbInformation.setCategoryType(fromGbInformation.getCategoryType());
+                List<CategoryDefinition> categories = toGbInformation.getCategories();
+                categories.add(toCategoryDefinition);
+                gradebookService.updateGradebookSettings(toGradebookId, toGbInformation);
+            }
+
+            Optional<CategoryDefinition> optional
+                = gradebookService.getCategoryDefinitions(toGradebookId)
+                    .stream()
+                    .filter(cd -> cd.getName().equals(categoryName)).findFirst();
```
Comment (ern) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: same findAny()

File: edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java [hunk 3]
```diff
@@ -228,6 +228,20 @@ public Object doInHibernate(final Session session) throws HibernateException {
 		}
 	}
 
+	public org.sakaiproject.service.gradebook.shared.Assignment getExternalAssignment(final String gradebookUid, final String externalId)
```
Comment (ern) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java]: You probably want to use the GradebookExternalAssessmentService when dealing with external assessments. As you have witnessed its strange the way these services are separated!

Comment (adrianfish) [reply edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java]: I think I'd rather just leave it in there. The returned type is different to the existing (private) external service method and the conversion method is in gradebookservice. It's really messy.

**Pull request merged**